### PR TITLE
feat: add String::write_utf8_uninit

### PR DESCRIPTION
**Pull request by Nugine**
This commit allows writing data to an uninitialized buffer.

Related:
+ https://github.com/denoland/deno/issues/15020

### CODE REVIEW

File: src/string.rs [hunk 1]
```diff
@@ -254,6 +255,22 @@ impl String {
     buffer: &mut [u8],
     nchars_ref: Option<&mut usize>,
     options: WriteOptions,
+  ) -> usize {
+    let buffer = unsafe {
```
Comment (aapoalas) [inline src/string.rs:259]: Can this method even be called in a safe way except for already safe buffers?

ie. Is this here only to not break the build and to enable a graceful transition into the `_uninit` API?

Comment (Nugine) [reply src/string.rs:259]: Yes. Adding a new method will not break the build.

There is another way:

```rust
 pub fn write_utf8(
    &self,
    scope: &mut Isolate,
    buffer: OutBuf<'_>
    nchars_ref: Option<&mut usize>,
    options: WriteOptions,
  ) -> usize
```

where the `OutBuf` is something like [base64-simd::OutBuf](https://docs.rs/base64-simd/0.6.2/base64_simd/struct.OutBuf.html)

**Comment by bartlomieju** [comment]
@Nugine could you rebase the PR and make the tests pass? I'm still interested in landing this PR

**Comment by Nugine** [comment]
>  1 workflow awaiting approval
> First-time contributors need a maintainer to approve running workflows

**Review by littledivy — APPROVED** [review]
LGTM, Thank you!

**Pull request merged**